azure-proto-core-test/SystemAssignedIdentityTests.cs

```diff
@@ -0,0 +1,348 @@
+﻿using azure_proto_core.Resources;
+using NUnit.Framework;
+using System;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using System.Text.Json;
+
+namespace azure_proto_core_test
+{
+    public class SystemAssignedIdentityTests : SystemAssignedIdentity
+    {
+
+        [TestCase(null, null, null, null, true)]
+        [TestCase("72f988bf-86f1-41af-91ab-2d7cd011db47", "de29bab1-49e1-4705-819b-4dfddceaaa98", "72f988bf-86f1-41af-91ab-2d7cd011db47", "de29bab1-49e1-4705-819b-4dfddceaaa98", false)]
+        public void CompareToZero(string tenantId1, string principalId1, string tenantId2, string principalId2, bool isNull)
```

Comment:
don't think  we need the extra isNull parameters, we can check the tenantId1 param itself.  This allows us to compare null to not null